src/graphql/resolvers.js

```diff
@@ -58,6 +99,78 @@ export default {
         discountCode: errors.length === 0 ? code : null,
         errors
       };
+    },
+
+    createContributor: async (_, { input }) => {
+      const shopifyCustomerID = await createShopifyCustomer(input);
+
+      const [
+        { totalContributions: contributionCount, pullRequests }
+      ] = await Promise.all([
+        getContributorInfo(input.githubUsername),
+        prisma.createContributor({
+          githubUsername: input.githubUsername,
+          email: input.email,
+          shopifyCustomerID
+        })
+      ]).catch(error => {
+        // @todo: more useful error logging
+        throw new ApolloError(error.message);
```

Comment:
Possible errors:
1. Prisma complains because we have a duplicate value for a unique field (meaning any duplicate value for any field stored in Prisma)
2. GitHub is down or the API rate limits us, meaning `getContributorInfo` fails
3. Prisma is down, so the `prisma.createContributor` call fails

Catching "_____ is down" would just be a "please try again" situation, so we can probably look for the first problem (duplicate entries in Prisma) and return a custom message there, then default to: "There was a problem creating your contributor account. Please try again."

Reply:
This is kind of in place. We pass through original errors if it doesn't meet one of our specific errors, which will help us debug but may result in some ugly errors. For now maybe we should just keep an eye on it? Maybe add Sentry?